src/tools/perf/libperf.c

```diff
@@ -170,10 +247,71 @@ void ucx_perf_test_start_clock(ucx_perf_context_t *perf)
     perf->prev.time         = perf->start_time;
 }
 
+static ucs_status_t setup_cuda(ucx_perf_context_t *perf, int group_index)
+{
+    int         dev_count;
+    CUdevice    device;
+    CUresult    err;
+
+    if (perf->ctx_set) return UCS_OK;
+
+    err = cuInit(0);
```

Comment:
any reason for using driver APIs?

Reply:
Because builds have dependency on just one library (libcuda) and not libcudart. I've used this throughout the PR actually.

Reply:
hmm,  cudart is already detected in the configuration. And I have been stick to cudart mostly whenever it is possible.